data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/compression/AutomaticCompressionEngineTest.java

```diff
@@ -71,4 +73,33 @@ void createInputStream_with_automatic_and_compressed_should_return_instance_of_G
         assertThat(inputStream, instanceOf(GzipCompressorInputStream.class));
         assertThat(inputStream.readAllBytes(), equalTo(testStringBytes));
     }
+
+    @Test
+    void createInputStream_with_automatic_and_compressed_should_return_instance_of_SnappyInputStream() throws IOException {
+         snappyTest(".snappy");
+    }
+
+    @Test
+    void createInputStream_with_automatic_and_compressed_should_return_instance_of_SnappyInputStreamParquet() throws IOException {
+        snappyTest(".snappy.parquet");
+    }
+    public void snappyTest(String fileExtension) throws IOException {
```

Comment:
You can make this a parameterized test.

```
@ParameterizedTest
@ValueSource(strings = { ".snappy", ".snappy.parquet"}
void createInputStream_with_automatic_and_compressed_should_return_instance_of_SnappyInputStream(final String fileExtension) {
  // The code from inside `snappyTest`.
}
```

Reply:
Addressed and now Junit has parameterized test